### PR TITLE
Fix reward space and add reward bound tests

### PR DESCRIPTION
**Pull request by LucasAlegre**

### CODE REVIEW

File: mo_gymnasium/envs/resource_gathering/resource_gathering.py [hunk 1]
```diff
@@ -30,9 +30,9 @@ class ResourceGathering(gym.Env, EzPickle):
 
     ## Reward Space
     The reward is 3-dimensional:
-    - 0: +1 if returned home with gold, else 0
-    - 1: +1 if returned home with diamond, else 0
-    - 2: -1 if killed by an enemy, else 0
+    - 0: -1 if killed by an enemy, else 0
```
Comment (ffelten) [inline mo_gymnasium/envs/resource_gathering/resource_gathering.py:33]: So this was buggy?

File: tests/test_envs.py [hunk 2]
```diff
@@ -97,6 +98,19 @@ def test_env_determinism_rollout(env_spec: EnvSpec):
...
 
+def _test_reward_bounds(env: gym.Env):
+    """Test that the reward bounds are respected."""
+    assert env.reward_dim is not None
+    assert env.reward_space is not None
+    env.reset()
+    for _ in range(NUM_STEPS):
+        action = env.action_space.sample()
+        _, reward, terminated, truncated, _ = env.step(action)
+        assert env.reward_space.contains(reward)
```
Comment (ffelten) [inline tests/test_envs.py:109]: assert reward shape and dtype?

**Review by ffelten — COMMENTED** [review]
🤝 👍

File: mo_gymnasium/envs/resource_gathering/resource_gathering.py (hunk 1 above)
Comment (LucasAlegre) [reply mo_gymnasium/envs/resource_gathering/resource_gathering.py:33]: Only the documentation

File: tests/test_envs.py (hunk 2 above)
Comment (LucasAlegre) [reply tests/test_envs.py:109]: .contains already takes care of it

**Review by ffelten — APPROVED** [review]

**Pull request merged**